src/modules/esl-popup/core/esl-popup.ts

```diff
@@ -121,6 +124,21 @@ export class ESLPopup extends ESLToggleable {
     return RTLUtils.isRtl(this) ? 1 - ratio : ratio;
   }
 
+  /** Moves popup into document.body */
+  protected moveToBody(): void {
+    const {parentNode, $placeholder} = this;
+    if (parentNode && parentNode !== document.body) {
+      if ($placeholder) {
+        // to be safe and prevent leaks
+        $placeholder.parentNode?.removeChild($placeholder);
+      }
+      // replace this with placeholder element
+      this.$placeholder = ESLPopupPlaceholder.from(this);
```

Comment:
Cosmetic: do we need to recreate elements? In case we going to have only a single instance per popup it makes sense to create a lazy getter. That simplifies our work with DOM because then 
`parentNode.replaceChild(this.$placeholder, this);`
Will work perfectly independently of preconditions.

Point to discuss: behavior on popup disconnection ... (keeping in mind that move also producing disconnection)

Reply:
Created the [issue](https://github.com/exadel-inc/esl/issues/795) to investigate behavior and solve some unusual cases.

Reply:
Thanks!